MSD: Update page headers for better mobile layout.

StevenDufresne opened the pull request:
Related to: #106954
Part of DOTCOM-15116

## Proposed Changes

This PR updates PageHeader to fold better on small screens and modifies subsequent views to improve layout.

| Small (@430px)| Medium (@768px) |
|--------|--------|
| <img width="1290" height="2796" alt="calypso localhost_3000_v2_me_profile(iPhone 14 Pro Max)" src="https://github.com/user-attachments/assets/056ce58f-9b5f-4040-99e9-a16bfe98af13" /> | <img width="1536" height="2048" alt="calypso localhost_3000_v2_me_profile(iPad Mini)" src="https://github.com/user-attachments/assets/4cd60f0a-d3ad-4dd8-b58a-7664696826fe" /> |
| <img width="1290" height="2796" alt="calypso localhost_3000_v2_me_profile(iPhone 14 Pro Max) (1)" src="https://github.com/user-attachments/assets/d7de4513-6764-4dd3-95c2-72ab090dea50" /> | <img width="1536" height="2048" alt="calypso localhost_3000_v2_me_profile(iPad Mini) (5)" src="https://github.com/user-attachments/assets/8aa40852-4588-4d28-ada2-6a2481f5bcb1" /> |
| <img width="1290" height="2796" alt="calypso localhost_3000_v2_me_profile(iPhone 14 Pro Max) (2)" src="https://github.com/user-attachments/assets/2f210055-14ec-4ac0-939a-b95d34ba9dcd" /> | <img width="1536" height="2048" alt="wpcalypso wordpress com_v2_sites_abstracting blog_logs_activity(iPad Mini)" src="https://github.com/user-attachments/assets/7000906e-af55-41c8-9cd4-66b2f2b60de0" /> |
| <img width="1290" height="2796" alt="calypso localhost_3000_v2_me_profile(iPhone 14 Pro Max) (3)" src="https://github.com/user-attachments/assets/4ec8c4c0-553a-42ff-b852-b1e3e5db587a" /> | <img width="1536" height="2048" alt="calypso localhost_3000_v2_me_profile(iPad Mini) (3)" src="https://github.com/user-attachments/assets/706262ad-8eae-472a-8d41-05b33d754789" /> |
| <img width="1290" height="2796" alt="calypso localhost_3000_v2_me_profile(iPhone 14 Pro Max) (4)" src="https://github.com/user-attachments/assets/5756284e-919f-4e25-91dc-c73ef2ce7ff3" /> | <img width="1536" height="2048" alt="calypso localhost_3000_v2_me_profile(iPad Mini) (2)" src="https://github.com/user-attachments/assets/95af8b39-7ba3-4b25-8d4c-2e80c5a96d7e" /> | 
| <img width="1290" height="2796" alt="wpcalypso wordpress com_v2_sites_abstracting blog_logs_activity(iPhone 14 Pro Max)" src="https://github.com/user-attachments/assets/796aaa60-3571-4f8a-8a0e-65591ffd4552" /> | <img width="1536" height="2048" alt="calypso localhost_3000_v2_me_profile(iPad Mini) (1)" src="https://github.com/user-attachments/assets/55e0a4bd-c896-4a83-a1ef-68f5ddecf897" />  |
|  <img width="1290" height="2796" alt="wpcalypso wordpress com_v2_sites_abstracting blog_logs_activity(iPhone 14 Pro Max) (1)" src="https://github.com/user-attachments/assets/3278311d-922b-479a-ba93-d8edd2136710" /> | <img width="1536" height="2048" alt="calypso localhost_3000_v2_me_profile(iPad Mini) (4)" src="https://github.com/user-attachments/assets/cedbfb83-b4d0-4de3-8444-ac51171776ee" /> |


## Testing Instructions

- Visit `/v2/`.
- Click around the desktop using a Mobile screen size preset and a tablet preset. 
- Pay close attention to how the PageHeader actions are presented.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

StevenDufresne committed:
Update spacing between elements.

StevenDufresne committed:
Make the font-size smaller on small screens.

StevenDufresne committed:
Expand the actions full on mobile.

StevenDufresne committed:
Update spacing.

StevenDufresne committed:
Remove extra classname.

StevenDufresne committed:
Fix the flex for the buttons.

StevenDufresne committed:
Add width auto to force HStack to take up its full space.

StevenDufresne committed:
Fix logs and monitoring.

StevenDufresne committed:
Get the expanding working.

StevenDufresne committed:
Update performance page header.

StevenDufresne committed:
If a page doesn't have actions we can't add the hstack or we'll get an extra gap.

StevenDufresne committed:
remove empty newline.

StevenDufresne commented on client/dashboard/components/section-header/index.tsx:
```diff
@@ -36,8 +37,8 @@ export const SectionHeader = ( {
-				<HStack justify="space-between" alignment="center" spacing={ 4 } wrap>
-					<VStack>
+				<HStack justify="space-between" alignment="center" spacing={ 6 } wrap>
+					<VStack style={ { flex: '1000 1 auto' } }>
```
This feels like magic. 

What this does is tell the browser to extend this container as much as you can; 1000 times more! 

This, combined with `{ flex: '1 1 auto' }` on the ButtonStack makes the button stack go full width once it wraps.

There may be a more logical way to do this that's escaping me at the moment.

StevenDufresne replied:
@arthur791004 usually has good advice for these issues. What do you think?

StevenDufresne replied:
This actually solves this problem too: https://github.com/Automattic/wp-calypso/pull/106898#issuecomment-3497012713

taipeicoder replied:
Hmmm how about forcing the wrap instead using breakpoints?

StevenDufresne replied:
Yeah, I orginally started with breakpoints, but for one action PageHeaders, it starts to look weird if you stack it too early. So pages like "Scans" and "Sites" with one action button are stacked early because "Performance" and "Backups," which include inputs, **need** to stack early.

I guess we could do it when there are multiple actions, or long actions, but we're getting into several conditionals that would make it unpredictable or configuration-heavy.

taipeicoder replied:
> I guess we could do it when there are multiple actions, or long actions, but we're getting into several conditionals that would make it unpredictable or configuration-heavy.

Agreed. I think this works for now if using breakpoints doesn't feel right.

fushar replied:
> This actually solves this problem too: https://github.com/Automattic/wp-calypso/pull/106898#issuecomment-3497012713

Hmm I'm still seeing the issue on my end:

<img width="1586" height="289" alt="image" src="https://github.com/user-attachments/assets/06bffbc8-6e12-4c69-b03e-8e99a9235c0c" />

arthur791004 replied:
What's the issue we wan to solve here by `flex: '1 1 auto'`?

StevenDufresne replied:
> What's the issue we wan to solve here by flex: '1 1 auto'?

When the buttons wrap, we want them to expand the full width of their container.

StevenDufresne replied:
I think I'm ready to merge this. I'm okay with a follow-up if a better idea presents itself.

StevenDufresne commented on client/dashboard/components/section-header/index.tsx:
```diff
@@ -48,15 +49,17 @@ export const SectionHeader = ( {
-					>
-						{ actions }
-					</ButtonStack>
+					{ !! actions && (
```
I'm making this conditional otherwise, there is an empty gap for containers with no actions. This doesn't impact the transition animation we currently have, and likely from when we had more advanced transitions. We can come back to fix this in the future if we want.

taipeicoder commented:
Hmmm it doesn't seem related to this PR. Try starting on /v2/emails then navigate to site-level tabs. There's a rather aggressive `.dashboard-section-header__actions` rule being inserted. 😕

StevenDufresne committed:
Remove css styles and change variant to match other sites.

StevenDufresne commented:
I've updated the email header, removing the style. I also changed the "Add email forwarder" from a link to a secondary button. That fixes the alignment and matches what we're doing elsewhere.

| Before | After |
|--------|--------|
| <img width="994" height="518" alt="Screenshot 2025-11-11 at 10 28 26 AM" src="https://github.com/user-attachments/assets/2846e5d8-4e68-442b-b37a-15b5abc80f36" /> | <img width="1031" height="512" alt="Screenshot 2025-11-11 at 10 28 33 AM" src="https://github.com/user-attachments/assets/d4975204-847f-4ec6-8782-9e92414bfd39" /> |

StevenDufresne commented:
Looking at active upgrades [the code](https://github.com/Automattic/wp-calypso/blob/e2e3cc293e0f08fed751be43fa738cec29797286/client/dashboard/me/billing-purchases/purchase-settings/index.tsx#L237), I see :
- There's an upgrade button and that action is duplicated as the first item in the three-dot menu.
  - This is conditional, sometimes, we only get the three-dot menu. 
- The three-dot menu seems to have a max of 2 item:
   - Upgrade
   - Renew 

I would prefer we actually remove that three-dot menu entirely, and add the two actions `[Upgrade] [Renew]` as buttons. Maybe there's a plan for more items? Maybe it's because of the limited space ...  cc @michaeldcain @sirbrillig 


| Before | After |
|--------|--------|
| <img width="863" height="207" alt="Screenshot 2025-11-11 at 11 26 10 AM" src="https://github.com/user-attachments/assets/e2d8d9a1-0c32-4f10-a6ae-7f593fee03c2" /> | <img width="705" height="176" alt="Screenshot 2025-11-11 at 11 27 33 AM" src="https://github.com/user-attachments/assets/d6beaabc-9c4f-4999-b2ee-6b045978997b" /> | 


If it is because of limited space, I find the "upgrade" button to be weird. Upgrading doesn't seem like the primary action when visiting your active upgrades. I would do that below, where we can make a good case for upgrading. This feels like it's just thrown in. 

| Idea | 
|--------|
| <img width="2606" height="3204" alt="calypso localhost_3000_v2_me_billing_purchases_25759686 (1)" src="https://github.com/user-attachments/assets/9aac9253-4ef7-453b-a25e-783a8a5c629d" /> |

StevenDufresne committed:
Space out menu on billing.

fushar commented on client/dashboard/app-ciab/style.scss:
```diff
@@ -41,7 +41,7 @@
 	--dashboard-h1__font-size: 40px;
 	--dashboard-h1__font-weight: 400;
-	--dashboard-h1__line-height: 54px;
+	--dashboard-h1__line-height: normal;
```
If it's `normal`, would it make more sense to just not specify it at all 🤔

StevenDufresne replied:
It's a good question. I'm assuming that being able to control this is an important complement to `--dashboard-h1__font-size` otherwise it never would have been added. For that reason, I didn't remove it.

arthur791004 commented on client/dashboard/emails/layout.tsx:
```diff
@@ -20,7 +20,7 @@ export const Layout = ( { children }: PropsWithChildren ) => {
 							<Button
 								className="emails__add-email-forwarder"
-								variant="link"
+								variant="secondary"
```
It's intented... See https://github.com/Automattic/wp-calypso/pull/106069#discussion_r2382267437 🫠

StevenDufresne replied:
Thanks for pointing that out. But I'll make the choice on this one. Let's prefer consistency.

lucasmendes-design commented:
LGTM! 🌟
